handler/src/main/java/io/netty/handler/ssl/SniHandler.java

```diff
@@ -362,51 +132,6 @@ protected void replaceHandler(ChannelHandlerContext ctx, String hostname, SslCon
         }
```

Comment:
I have but as per previous  https://github.com/netty/netty/pull/6623#discussion_r113187618 there's likely no "value" to it in the sense that somebody who extends `AbstractSniHandler` is probably comfortable with the state management and is doing it on purpose.

@normanmaurer @Scottmitch But in regards to leaks, can you guys take a look at https://github.com/netty/netty/blob/4.1/handler/src/main/java/io/netty/handler/ssl/SniHandler.java#L330 or this PR's SniHandler (line 100).

If `replaceHandler(...)` throws an Exception we'll leak the `SslContext` if I'm not wrong?! :thinking: